test/extensions/filters/http/json_to_metadata/filter_test.cc

```diff
@@ -0,0 +1,1328 @@
+#include <memory>
```

Comment:
Recommend adding some integration test using the filter as it'll allow us to test the filter usage as when it's in the filter chain

Reply:
done. While I wrote my first filter and I have a bug in the factory code so which is not able to test in config/filter_test. Thanks for suggestion!